src/appleseed.bench/utility/backendapi.cpp

```diff
@@ -0,0 +1,239 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2019 Francois Beaune, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "backendapi.h"
+
+// appleseed.bench headers.
+#include "utility/encryptionkey.h"
+
+// appleseed.foundation headers.
+#include "foundation/platform/compiler.h"
+
+// Qt headers.
+#include <QByteArray>
+#include <QFile>
+#include <QIODevice>
+#include <QJsonObject>
+#include <QJsonValue>
+#include <QNetworkRequest>
+#include <QTextStream>
+#include <QUrl>
+#include <QVariant>
+
+// Standard headers.
+#include <cassert>
+#include <stdexcept>
+
+namespace appleseed {
+namespace bench {
+
+//
+// Setup:
+//
+//   1. Generate a new API key:
```

Comment:
Shouldn't we start to make a proper *dev* documentation somewhere outside of the code ? Or do you think it's better to keep it here ?

Reply:
We already have the wiki which is well suited to developer documentation.

But for this particular bit of docs, I feel it's best to keep it here: that way it's versioned together with the code.